shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+
+                let blockedKey = "blockedCookie";
+                let allowedKey = "allowedCookie";
+                let newValue = newId();
+                
+                // Try to set blocked cookie while disabled - should not be cached
+                let blockedResult = manager.set(blockedKey, newValue);
+                Assert.equal(false, blockedResult, "Setting blocked cookie should return false");
+                Assert.equal("", manager.get(blockedKey), "Should not return cached value for blocked cookie");
+                
+                // Set allowed cookie while disabled - should be cached
+                let allowedResult = manager.set(allowedKey, newValue);
+                Assert.equal(true, allowedResult, "Setting allowed cookie should return true");
+                Assert.equal(newValue, manager.get(allowedKey), "Should return cached value for allowed cookie");
+
+                // Enable cookies - only allowed cookie should be flushed
+                manager.setEnabled(true);
+                
+                Assert.equal(undefined, this._testCookies[blockedKey], "Blocked cookie should not be in storage");
+                Assert.equal(newValue + "; path=/", this._testCookies[allowedKey], "Allowed cookie should be in storage");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Write cookies after being enabled - handle ignored cookies",
+            test: () => {
+                let cookieCfg: ICookieMgrConfig = objExtend(true, {}, this._cookieMgrCfg);
+                cookieCfg.ignoreCookies = ["ignoredCookie"];
+                
+                let manager = createCookieMgr({
+                    cookieCfg: cookieCfg
+                });
+
+                // Start with cookies disabled
+                manager.setEnabled(false);
+
+                let ignoredKey = "ignoredCookie";
+                let allowedKey = "allowedCookie";
+                let newValue = newId();
+                
+                // Try to set ignored cookie while disabled - should not be cached
+                let ignoredResult = manager.set(ignoredKey, newValue);
+                Assert.equal(false, ignoredResult, "Setting ignored cookie should return false");
+                Assert.equal("", manager.get(ignoredKey), "Should not return cached value for ignored cookie");
+                
+                // Set allowed cookie while disabled - should be cached
+                let allowedResult = manager.set(allowedKey, newValue);
+                Assert.equal(true, allowedResult, "Setting allowed cookie should return true");
+                Assert.equal(newValue, manager.get(allowedKey), "Should return cached value for allowed cookie");
+
+                // Enable cookies - only allowed cookie should be flushed
+                manager.setEnabled(true);
+                
+                Assert.equal(undefined, this._testCookies[ignoredKey], "Ignored cookie should not be in storage");
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect